File: src/main/java/com/telegram/folobot/service/handlers/ReplyHandler.java
package com.telegram.folobot.service.handlers;

import com.telegram.folobot.FoloBot;
import com.telegram.folobot.service.MessageService;
import com.telegram.folobot.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import org.telegram.telegrambots.meta.api.methods.BotApiMethod;
import org.telegram.telegrambots.meta.api.objects.Update;

import java.util.SplittableRandom;

import static com.telegram.folobot.ChatId.isAndrew;

@Component
@RequiredArgsConstructor
public class ReplyHandler {
    private final UserService userService;
    private final MessageService messageService;

    /**
     * Ответ на обращение
     *
     * @param update {@link Update}
     * @return {@link BotApiMethod}
     */
    public BotApiMethod<?> handle(Update update) {
        // Cообщение в чат
        String text = update.getMessage().getText().toLowerCase();
        if (text.contains("привет") || new SplittableRandom().nextInt(100) < 20) {
            String userName = userService.getFoloUserName(update.getMessage().getFrom());
            messageService.sendChatTyping(update);
            if (isAndrew(update.getMessage().getFrom())) {
                return messageService
                        .buildMessage("Привет, моя сладкая бориспольская булочка!", update, true);
            } else {
                return messageService
                        .buildMessage("Привет, уважаемый фолофил " + userName + "!", update, true);
            }
        }
        return null;
    }
}

File: src/main/java/com/telegram/folobot/service/handlers/ReplyHandler.kt
package com.telegram.folobot.service.handlers

import com.telegram.folobot.ChatId.Companion.isAndrew
import com.telegram.folobot.service.MessageService
import com.telegram.folobot.service.UserService
import org.springframework.stereotype.Component
import org.telegram.telegrambots.meta.api.methods.BotApiMethod
import org.telegram.telegrambots.meta.api.objects.Update
import java.util.*

@Component
class ReplyHandler(
    private val userService: UserService,
    private val messageService: MessageService
) {

    /**
     * Ответ на обращение
     *
     * @param update [Update]
     * @return [BotApiMethod]
     */
    fun handle(update: Update): BotApiMethod<*>? {
        // Cообщение в чат
        val text = update.message.text.lowercase(Locale.getDefault())
        if (text.contains("привет") || SplittableRandom().nextInt(100) < 20) {
            val userName = userService.getFoloUserName(update.message.from)
            messageService.sendChatTyping(update)
            return if (isAndrew(update.message.from)) {
                messageService
                    .buildMessage("Привет, моя сладкая бориспольская булочка!", update, true)
            } else {
                messageService
                    .buildMessage("Привет, уважаемый фолофил $userName!", update, true)
            }
        }
        return null
    }
}